kafka/src/main/java/com/metamx/tranquility/kafka/writer/TranquilityEventWriter.java

```diff
@@ -0,0 +1,231 @@
+import com.metamx.common.logger.Logger;
+import com.metamx.tranquility.beam.Beam;
+import com.metamx.tranquility.beam.BeamPacketizer;
+import com.metamx.tranquility.beam.BeamPacketizerListener;
+import com.metamx.tranquility.beam.ClusteredBeamTuning;
+import com.metamx.tranquility.druid.DruidBeams;
+import com.metamx.tranquility.druid.DruidDimensions;
+import com.metamx.tranquility.druid.DruidLocation;
+import com.metamx.tranquility.druid.DruidRollup;
+import com.metamx.tranquility.druid.DruidSpatialDimension;
+import com.metamx.tranquility.kafka.model.IngestionSpec;
+import com.metamx.tranquility.kafka.model.TranquilityConfig;
+import com.metamx.tranquility.kafka.util.IngestionSpecHelper;
+import com.metamx.tranquility.typeclass.Timestamper;
+import io.druid.data.input.impl.SpatialDimensionSchema;
+import io.druid.segment.indexing.DataSchema;
+import org.apache.curator.RetryPolicy;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.joda.time.DateTime;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An EventWriter that pushes events to Druid through Tranquility.
+ */
+public class TranquilityEventWriter implements EventWriter
+{
+  private static final Logger log = new Logger(TranquilityEventWriter.class);
+  private static final ObjectMapper mapper = new ObjectMapper();
+  private static final RetryPolicy retryPolicy = new ExponentialBackoffRetry(1000, 500, 30000);
+
+  private final CuratorFramework curator;
+  private final String indexService;
+  private final String discoveryPath;
+  private final String specFile;
+  private final String topic;
+
+  private BeamPacketizer<Map<String, Object>> beamPacketizer;
+  private String timestampColumn;
+
+  public TranquilityEventWriter(String zkConnect, String indexService, String discoveryPath,
+                                String specFile, String topic) {
+    this.indexService = indexService;
+    this.discoveryPath = discoveryPath;
+    this.specFile = specFile;
+    this.topic = topic;
+
+    curator = CuratorFrameworkFactory.builder().connectString(zkConnect).retryPolicy(retryPolicy).build();
+  }
+
+  @Override
+  public void init(final EventWriterListener listener)
+  {
+    final IngestionSpec ingestionSpec = IngestionSpecHelper.parseSpecFile(specFile);
+    final DataSchema schema = ingestionSpec.getDataSchema();
+    final TranquilityConfig tranquilityConfig = ingestionSpec.getConfig();
+    final String dataSource = tranquilityConfig.useTopicAsDataSource() ? topic : schema.getDataSource();
+    final String firehosePattern = "tranquility-kafka:" + indexService + ":" + dataSource + ":%s";
+
+    timestampColumn = schema.getParser().getParseSpec().getTimestampSpec().getTimestampColumn();
+
+    final Timestamper<Map<String, Object>> timestamper = new Timestamper<Map<String, Object>>()
```

Comment:
This will probably happen on its own if you end up pulling out the configuring code from server's `Main` into core, as it does do this.